### PR TITLE
V434 mem leak repairs

### PR DESCRIPTION
**Pull request by baallan**
This fixes several memory leaks in configuration and ldms_ls handling paths and adds the test scripts that uncovered them.
Each bug fix is in its own commit.

### CODE REVIEW

File: lib/src/ovis_json/ovis_json_parser.y [hunk 1]
```diff
@@ -109,9 +109,11 @@ attr_list: /* empty */ { $$ = new_dict_val(); }
 	 | string ':' value {
 	     json_entity_t e = new_dict_val();
 	     $$ = add_dict_attr(e, $1, $3);
+	     json_entity_free($1);
```
Comment (tom95858) [inline lib/src/ovis_json/ovis_json_parser.y:112]: @nichamon can you please confirm this leak?

File: lib/src/ovis_json/ovis_json_parser.y [hunk 2]
```diff
@@ -109,9 +109,11 @@ attr_list: /* empty */ { $$ = new_dict_val(); }
...
 	 }
 	 | attr_list ',' string ':' value {
 	     $$ = add_dict_attr($1, $3, $5);
+	     json_entity_free($3);
```
Comment (nichamon) [inline lib/src/ovis_json/ovis_json_parser.y:116]: @tom95858 With the same reason above, this is also legit.

**Comment by tom95858** [comment]
Thanks @baallan for finding and fixing this.

File: lib/src/ovis_json/ovis_json_parser.y (hunk 2 above)
Comment (baallan) [reply lib/src/ovis_json/ovis_json_parser.y:116]: @nichamon is there an input file that exercises the entire grammar (lex and yacc) that we can test routinely as the config tools evolve?

**Pull request merged**